File: 14-Longest-Common-Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        strs.sort()

        prefix = ""
        """
        The zip() function is versatile and useful for combining 
        and manipulating data from     
        multiple iterables in Python. It's commonly 
        used in scenarios where you need to process 
        corresponding elements from different sequences together
        """
        for c1, c2 in zip(strs[0], strs[-1]):
            if c1 == c2: prefix += c1
            else: break
        return prefix

File: 14-Longest-Common-Prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let prefix = strs[0];
    for (let i = 0; i < strs.length; i++) {
        while(strs[i].indexOf(prefix) !== 0) {
            prefix = prefix.slice(0, prefix.length - 1);

            if (prefix == \\) {
                return \\
            }
        }

    }
    return prefix
};